<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1343</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ZMQ Test Tool</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="labelHost">
          <property name="text">
           <string>Host:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditHost">
          <property name="statusTip">
           <string>Input Host IP Address</string>
          </property>
          <property name="text">
           <string>127.0.0.1</string>
          </property>
          <property name="placeholderText">
           <string>Input Host IP Address Here</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelPublishPort">
          <property name="text">
           <string>Publish Port:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBoxPortPublish">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="statusTip">
           <string>Input Publish Port</string>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="value">
           <number>9445</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelSubscribePort">
          <property name="text">
           <string>Subscribe Port:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBoxPortSubscribe">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="statusTip">
           <string>Input Subscribe Port</string>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="value">
           <number>9446</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="buttonDefault">
        <property name="text">
         <string>Default</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/images.qrc">
          <normaloff>:/images/clearbytes.png</normaloff>:/images/clearbytes.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonStop">
        <property name="statusTip">
         <string>Stop Publishing Message</string>
        </property>
        <property name="text">
         <string>Disconnect</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/images.qrc">
          <normaloff>:/images/stop.png</normaloff>:/images/stop.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonStart">
        <property name="statusTip">
         <string>Start Publishing Message</string>
        </property>
        <property name="text">
         <string>Connect</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/images.qrc">
          <normaloff>:/images/open.png</normaloff>:/images/open.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_9">
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    background-color: qlineargradient(
        x1: 0, y1: 0,
        x2: 0, y2: 1,
        stop: 0 #E0E0E0,
        stop: 1 #FFFFFF
    );

    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 5ex;
}

QGroupBox::title {
    subcontrol-origin: margin;
    padding: 5px 0px;
}</string>
        </property>
        <property name="title">
         <string>Conversation Window:</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="title">
            <string>Subscribe Topic Information</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="labelPublishTopic_4">
              <property name="text">
               <string>Topic Operation:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTabWidget" name="tabWidget">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <widget class="QWidget" name="tab">
               <attribute name="icon">
                <iconset resource="../resources/images.qrc">
                 <normaloff>:/images/loadfile.png</normaloff>:/images/loadfile.png</iconset>
               </attribute>
               <attribute name="title">
                <string>RawView</string>
               </attribute>
               <layout class="QVBoxLayout" name="verticalLayout_4">
                <item>
                 <widget class="QTextEdit" name="textView">
                  <property name="statusTip">
                   <string>Show Publish Messages</string>
                  </property>
                  <property name="placeholderText">
                   <string>Messages will shown here once the publish is triggered</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="tab_4">
               <attribute name="icon">
                <iconset resource="../resources/images.qrc">
                 <normaloff>:/images/wincom.ICO</normaloff>:/images/wincom.ICO</iconset>
               </attribute>
               <attribute name="title">
                <string>LogMessage</string>
               </attribute>
               <layout class="QVBoxLayout" name="verticalLayout_8">
                <item>
                 <widget class="QTextEdit" name="logMessage">
                  <property name="statusTip">
                   <string>Show Log Message</string>
                  </property>
                  <property name="placeholderText">
                   <string>Log message will shown here</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLabel" name="labelPublishFrequency_3">
                <property name="text">
                 <string>Update Frequency (Hz):</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBoxFrequency">
                <property name="maximum">
                 <number>200</number>
                </property>
                <property name="value">
                 <number>10</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="labelDisplayPublish">
                <property name="text">
                 <string>Display as:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="decDisplay">
                <property name="text">
                 <string>Decimal</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="hexDisplay">
                <property name="text">
                 <string>Hexadecimal</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="buttonClearAll">
                <property name="statusTip">
                 <string>Clear All Information Input</string>
                </property>
                <property name="text">
                 <string>Clear All</string>
                </property>
                <property name="icon">
                 <iconset resource="../resources/images.qrc">
                  <normaloff>:/images/edit-clear.png</normaloff>:/images/edit-clear.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="labelPublishMessageCount">
                <property name="text">
                 <string>Publish Message Count:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLCDNumber" name="lcdNumberPublish"/>
              </item>
              <item>
               <widget class="QLabel" name="labelSubscribeMessageCount">
                <property name="text">
                 <string>Subscribe Message Count:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLCDNumber" name="lcdNumberSubscribe"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="title">
            <string>Subscribe Topic List</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QLabel" name="labelPublishTopic_3">
              <property name="text">
               <string>Topic Table:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTableView" name="tableViewTopics"/>
            </item>
            <item>
             <widget class="QLabel" name="labelPublishTopic">
              <property name="text">
               <string>Topic Modification:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEditSubscribeTopic">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="statusTip">
               <string>Input Publish Topic Here</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="placeholderText">
               <string>Input Topic Here</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="buttonRemoveTopic">
                <property name="statusTip">
                 <string>Start Subscribing Message</string>
                </property>
                <property name="text">
                 <string>Remove</string>
                </property>
                <property name="icon">
                 <iconset resource="../resources/images.qrc">
                  <normaloff>:/images/cleanport.png</normaloff>:/images/cleanport.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="buttonAddTopic">
                <property name="statusTip">
                 <string>Start Subscribing Message</string>
                </property>
                <property name="text">
                 <string>Add</string>
                </property>
                <property name="icon">
                 <iconset resource="../resources/images.qrc">
                  <normaloff>:/images/add.png</normaloff>:/images/add.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="styleSheet">
       <string notr="true">QGroupBox {
    background-color: qlineargradient(
        x1: 0, y1: 0,
        x2: 0, y2: 1,
        stop: 0 #E0E0E0,
        stop: 1 #FFFFFF
    );

    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 5ex;
}

QGroupBox::title {
    subcontrol-origin: margin;
    padding: 5px 0px;
}</string>
      </property>
      <property name="title">
       <string>Public Message</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="labelPublishTopic_2">
           <property name="text">
            <string>Publish Topic:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditPublishTopic">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="statusTip">
            <string>Input Publish Message Here</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="placeholderText">
            <string>Input Publish Topic Here</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="labelPublishFrequency">
           <property name="text">
            <string>Publish Frequency (Hz):</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="publishFrequency">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>200</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBoxLoop">
           <property name="text">
            <string>Loop Publish</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="buttonPublishClearAll">
           <property name="statusTip">
            <string>Clear All Information Input</string>
           </property>
           <property name="text">
            <string>Clear All</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/images.qrc">
             <normaloff>:/images/edit-clear.png</normaloff>:/images/edit-clear.png</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="buttonSend">
           <property name="statusTip">
            <string>Start Publishing Message</string>
           </property>
           <property name="text">
            <string>Send</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/images.qrc">
             <normaloff>:/images/send.png</normaloff>:/images/send.png</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QLabel" name="labelPublishMessage">
           <property name="text">
            <string>Publish Message:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditPublishMessage">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="statusTip">
            <string>Input Publish Message Here</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="placeholderText">
            <string>Input Publish Message Here</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="iconLable">
        <property name="maximumSize">
         <size>
          <width>130</width>
          <height>40</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="../resources/images.qrc">:/images/logo-toggle.svg</pixmap>
        </property>
        <property name="scaledContents">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="jouav">
        <property name="text">
         <string>www.jouav.com</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="buttonQuit">
        <property name="text">
         <string>Quit</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/images.qrc">
          <normaloff>:/images/exit.png</normaloff>:/images/exit.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1343</width>
     <height>17</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/exit.png</normaloff>:/images/exit.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/message.png</normaloff>:/images/message.png</iconset>
   </property>
   <property name="text">
    <string>About ZMQ Test Tool</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="../resources/images.qrc"/>
 </resources>
 <connections/>
</ui>
